modules/hal_gigadevice/CMakeLists.txt

```diff
@@ -5,51 +5,64 @@ if(CONFIG_HAS_GD32_HAL)
 
 string(TOUPPER ${CONFIG_SOC} gd32_soc_uc)
 
-set(gd32_soc_dir     ${ZEPHYR_HAL_GIGADEVICE_MODULE_DIR}/${CONFIG_SOC_SERIES})
-set(gd32_cmsis_dir   ${gd32_soc_dir}/cmsis/gd/${CONFIG_SOC_SERIES})
+if(CONFIG_NUCLEI_CORE_VARIANT)
+  string(TOLOWER ${CONFIG_NUCLEI_CORE_VARIANT} gd32_soc_series_lc)
+  set(gd32_soc_dir     ${ZEPHYR_HAL_GIGADEVICE_MODULE_DIR}/${gd32_soc_series_lc})
```

Comment:
I think there is no needed to rework using gd32_soc_series_lc if we can add SOC_FAMILY_GD32_RISCV and change value at SOC_SERIES

Reply:
I'll apply it.

Reply:
If value of SOC_SERIES keep as "nuclei", this part also need to keep as original PR.
https://github.com/zephyrproject-rtos/zephyr/pull/34970#discussion_r757437888

Reply:
Restoring CONFIG_SOC_SERIES with SOC_SERIES changeing to gd32vf103.